spec/controllers/books_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+describe BooksController, type: :controller do
+  let(:user) { create(:user) }
+
+  before { login(user) }
+
+  describe 'GET #index' do
+    context 'When fetching all the books' do
+      let!(:books) { create_list(:book, 3) }
+
+      before do
+        get :index, params: { id: user.id }
+      end
+
+      it 'responses with the users books json' do
+        expected = ActiveModel::Serializer::CollectionSerializer.new(
+          books, each_serializer: BookSerializer
+        ).to_json
+        expect(JSON.parse(response.body)) =~ JSON.parse(expected)
```

Comment:
this operator `=~` is used when you want to match a regular expression, and I think you don't want that here

Reply:
ok, but you can access the page and then compare the jsons

Reply:
ok, but you can access the page and then compare the jsons